src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -541,9 +689,778 @@ private bool CheckSensorHandle()
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void ProviderSetRange(float minRange, float maxRange)
+        {
+            if (CheckProviderHandle())
+            { 
+                int error = Interop.SensorProvider.ProviderSetRange(_sensorProviderHandle, minRange, maxRange);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in setting range of sensor provider");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetRange Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Sets the resolution of sensor values to the sensor provider.
+        /// </summary>
+        /// <param name="resolution">
+        /// The resolution.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void ProviderSetResolution(float resolution)
+        {
+            if (CheckProviderHandle())
+            {
+                int error = Interop.SensorProvider.ProviderSetResolution(_sensorProviderHandle, resolution);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in setting resolution of sensor provider");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetResolution Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when a listener starts the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void SetProviderStartCb()
+        {
+            if (CheckProviderHandle())
+            {
+                _sensorProviderStartCallBack = (IntPtr _provider, Int64 _userData) =>
+                {
+                };
+
+                int error = Interop.SensorProvider.SetProviderStartCb(_sensorProviderHandle, _sensorProviderStartCallBack, IntPtr.Zero);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in setting start callback of sensor provider");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetStartCb Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
         }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when a sensor listener stops the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void SetProviderStopCb()
```

Comment:
What is purpose of this method?

Reply:
This API sets the callback function for getting called when provider stop. As per sensor spec need to set following 3 callbacks before calling AddProvider API.
1.  sensor_provider_start_cb()
2. sensor_provider_stop_cb(),
3. sensor_provider_set_interval_changed_cb()